### PR TITLE
home: switch to BusLocator-oriented helpers

### PR DESCRIPTION
**Pull request by vcaputo**
Remove some unnecessary verbosity, largely mechanical change with
no functional difference.

### CODE REVIEW

**Comment by vcaputo** [comment]
Assuming https://github.com/systemd/systemd/pull/15331 lands in a form amenable to this merging here's a bit of profitable use.

I don't need to run through all the code and do all the obvious wins here myself, in case anyone else wants in on the mechanical fun of deleting lots of lines.

**Comment by dnicolodi** [comment]
What about a macro like
```
#define DECLARE_BUS_CONVENIENCE_WRAPPERS(name, destination, path, interface) \
           int name##_call_method(                                      \
                   	sd_bus *bus,                                    \
                   	const char *member,                             \
                   	sd_bus_error *error,                            \
                   	sd_bus_message **reply,                         \
                   	const char *types) {                            \
                   va_list ap;                                          \
                   int r;                                               \
                   va_start(ap, types);                                 \
                   r = sd_bus_call_methodv(bus, destination, path, interface, member, error, reply, types, ap); \
                   va_end(ap);                                          \
                   return r;                                            \
           }                                                            \
           ...
```
that expands to definitions of the trivial wrappers with the common parameters filled in, instead?

**Comment by dnicolodi** [comment]
Another idea could be, instead of a `BusAddress` struct encapsulating destination, path, and interface, to use a `sd_bus_object` struct encapsulating bus, destination, path and interface. This spares one more argument and the semantic gets close to other libraries and messaging protocols. In this case the functions could use the `sd_bus_object_` prefix which read quite nicely (or `sd_bus_obj_` for maximum conciseness, but I like this less and I don't think three chars are worth the ugly abbreviation).

**Comment by keszybz** [comment]
This needs to be updated for #15331 now.

**Comment by poettering** [comment]
(and maybe rename BusAddress → BusLocator in this PR too, see discussion in #15331)

File: src/home/homectl.c [hunk 1]
```diff
@@ -65,6 +65,11 @@ static enum {
         EXPORT_FORMAT_STRIPPED,      /* strip "state" + "binding", but leave signature in place */
         EXPORT_FORMAT_MINIMAL,       /* also strip signature */
 } arg_export_format = EXPORT_FORMAT_FULL;
+static const BusAddress home_mgr = {
```
Comment (poettering) [inline src/home/homectl.c]: i'd place a an empty line before this

Comment (vcaputo) [reply src/home/homectl.c]: fixed

**Comment by vcaputo** [comment]
Updated to use BusLocator instead, and added the newline @poettering requested...

Wasn't sure if it was preferred to have a separate PR for the BusLocator rename, or just merge this to get both at once.  Up to you guys, I can always rebase.

**Comment by keszybz** [comment]
Looks nice, but centos ci fails with:
```
19:17:45     archlinux_systemd_ci: ../build/src/home/homectl.c:1633:21: error: implicit declaration of function 'bus_message_new_method_call' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
19:17:45     archlinux_systemd_ci:                 r = bus_message_new_method_call(bus, &m, &home_mgr, "UpdateHome");
19:17:45     archlinux_systemd_ci:                     ^
```

**Comment by vcaputo** [comment]
Well that's dumb, I've been doing plain compile tests here and they passed but looking at the code there's no helper for that one so apparently my local builds aren't building homectl.

I'll add a helper for that one.

**Comment by vcaputo** [comment]
Merging https://github.com/systemd/systemd/pull/15388 should bring in the needed helper, and yeah it turns out my local builds have homed disabled and it'd be a pita to get it enabled on this old debian box while bandwidth constrained due to unmet deps.

**Comment by vcaputo** [comment]
Pushed rebased on current master, hopefully centos ci is happy now!

**Comment by keszybz** [comment]
```
../src/home/homectl.c: In function ‘update_home’:
../src/home/homectl.c:1701:58: error: passing argument 3 of ‘bus_message_new_method_call’ from incompatible pointer type [-Werror=incompatible-pointer-types]
 1701 |                 r = bus_message_new_method_call(bus, &m, "ChangePasswordHome");
      |                                                          ^~~~~~~~~~~~~~~~~~~~
      |                                                          |
      |                                                          const char *
In file included from ../src/home/homectl.c:11:
../src/shared/bus-util.h:201:84: note: expected ‘const BusLocator *’ but argument is of type ‘const char *’
  201 | int bus_message_new_method_call(sd_bus *bus, sd_bus_message **m, const BusLocator *locator, const char *member);
      |                                                                  ~~~~~~~~~~~~~~~~~~^~~~~~~
../src/home/homectl.c:1701:21: error: too few arguments to function ‘bus_message_new_method_call’
 1701 |                 r = bus_message_new_method_call(bus, &m, "ChangePasswordHome");
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by vcaputo** [comment]
Fixed, and I think that's going to be all for my systemd contributions until I have time and bandwidth to update my environment and give these things more attention.

**Comment by vcaputo** [comment]
@keszybz how to force a rerun of CI tests?  I don't think they reran after last push

**Pull request merged**